test_zipp.py

```diff
@@ -416,3 +417,23 @@ def test_inheritance(self, alpharep):
         for alpharep in self.zipfile_alpharep():
             file = cls(alpharep).joinpath('some dir').parent
             assert isinstance(file, cls)
+
+    def test_can_pickle_string_path(self):
+        path_1 = zipp.Path("/path/to/a/file.zip")
+        path_1_pickle = pickle.dumps(path_1)
+        path_1_load = pickle.loads(path_1_pickle)
+        assert path_1.root == path_1_load.root
+        path_2 = zipp.Path("/path/to/a/file.zip", at="something.txt")
+        path_2_pickle = pickle.dumps(path_2)
+        path_2_load = pickle.loads(path_2_pickle)
+        assert path_2.root == path_2_load.root
+
+    def test_can_pickle_pathlib_path(self):
+        path_1 = zipp.Path(pathlib.Path("/path/to/a/file.zip"))
+        path_1_pickle = pickle.dumps(path_1)
+        path_1_load = pickle.loads(path_1_pickle)
+        assert path_1.root == path_1_load.root
+        path_2 = zipp.Path(pathlib.Path("/path/to/a/file.zip", at="something.txt"))
```

Comment:
This looks like a mistake. I don't think pathlib.Path accepts an `at` argument. Strange that the tests passed.

I've reported an issue upstream to track (https://github.com/python/cpython/issues/98094).

Reply:
Fixed in f945f0a4b9abab2449f5fd75d62e09cfe6dc35a9.